poller-lambdas/src/types.ts

```diff
@@ -0,0 +1,23 @@
+export type SecretValue = string; //TODO refine type
+
+export type PollerInput = string;
+
+export interface CorePollerOutput {
+	payloadForIngestionLambda: unknown; //TODO change to be the shared/agreed type the ingestion lambda expects
+}
+export type LongPollOutput = CorePollerOutput & { valueForNextPoll: string };
+
+export type FixedFrequencyPollOutput = CorePollerOutput & {
+	idealFrequencyInSeconds: number;
+};
+
+export type LongPollFunction = (
+	secret: SecretValue,
+	input: PollerInput,
```

Comment:
```suggestion
	input: PollerInput | undefined,
```

my current understanding is that input needs to be potentially undefined, to account for the case where a new deployment needs to bootstrap itself (i.e. where there's no pre-existing poller output to take in).

given that the SQS invocation needs to set some body value every time (as per https://github.com/guardian/newswires/pull/83/files#diff-fde74c4679830198a8758c56ef27dc2a92722727d787a4d36a93f82068f80af6R71) it looks like the poller-lambda wrapper will need to do something to distinguish between the case where there's a valid PollerInput and where there isn't

Reply:
We discussed this off-thread; because we want to only have one 'chain' of requests for a given poller at any given time (to avoid duplication of work). So bootstrapping a new chain of tasks in the SQS queue that drives the lambda should be done intentionally and therefore it may make sense to just pass the 'last known good value' manually when we start it. (This isn't a super clear description; I need to log off now but will try to revisit to clarify tomorrow.)